podcasts/IAPHelper.swift

```diff
@@ -224,21 +224,18 @@ extension IAPHelper {
         let formattedPrice = numberFormatter.string(from: offer.price)
         return formattedPrice ?? ""
     }
-
-    /// Returns the first product with a free trial
-    /// The priority order is set by the productIdentifiers array
-    /// - Returns: The product enum with a free trial or nil if there is no free trial
-    typealias FreeTrialDetails = (duration: String, pricing: String)
-    func getFirstFreeTrialDetails() -> FreeTrialDetails? {
-        guard
-            let product = getFirstFreeTrialProductId(),
-            let duration = localizedFreeTrialDuration(product),
-            let pricing = pricingStringWithFrequency(for: product)
-        else {
+    
+    /// Return the offer localized string for the end date of a product offer
+    /// - Parameter identifier: the product id that we want to check the offer
+    /// - Returns: the localized string for the end date of a product offer
+    func offerEndDate(_ identifier: IAPProductID) -> String? {
+        guard let offer = getFreeTrialOffer(identifier) else {
             return nil
         }
 
-        return (duration, pricing)
+        let date = offer.subscriptionPeriod.offerEndDate
+
+        return date?.formatted(date: .numeric, time: .omitted)
```

Comment:
Looking at the designs, it looks like the date displayed is in a longer form. Wdyt about changing this to long?

```suggestion
        return date?.formatted(date: .long, time: .omitted)
```